src/Polysemy/Internal/TH/Effect.hs

```diff
@@ -140,18 +140,20 @@ genFreer should_mk_sigs type_name = do
                          ]
   decs       <- traverse (genDec should_mk_sigs) cl_infos
 
-  let sigs = [ genSig <$> cl_infos | should_mk_sigs ]
+  let sigs = if should_mk_sigs then genSig <$> cl_infos else []
 
   return $ join $ def_mod_fi : sigs ++ decs
 
 ------------------------------------------------------------------------------
 -- | Generates signature for lifting function and type arguments to apply in
 -- its body on effect's data constructor.
-genSig :: ConLiftInfo -> Dec
+genSig :: ConLiftInfo -> [Dec]
 genSig cli
-  = SigD (cliFunName cli) $ quantifyType
-  $ ForallT [] (member_cxt : cliFunCxt cli)
-  $ foldArrows $ cliFunArgs cli ++ [sem `AppT` cliResType cli]
+  =  maybe [] ((:[]) . (`InfixD` cliFunName cli)) (cliFunFixity cli)
```

Comment:
I prefer `flip` instead of infix syntax, and `pure` instead of `(:[])`, as both trim down the amount of non-letters I need to read. Perhaps `maybe [] (pure . flip InfixD $ cliFunName cli)`?

Reply:
Yeah, probably better.